atecc608a_se.c

```diff
@@ -208,24 +247,68 @@ static psa_status_t atecc608a_asymmetric_sign(psa_key_slot_number_t key_slot,
     return status;
 }
 
+psa_status_t atecc608a_asymmetric_verify(psa_key_slot_number_t key_slot,
+                                         psa_algorithm_t alg,
+                                         const uint8_t *p_hash,
+                                         size_t hash_length,
+                                         const uint8_t *p_signature,
+                                         size_t signature_length)
+{
+    const uint16_t key_id = key_slot;
+    psa_status_t status = PSA_ERROR_GENERIC_ERROR;
+    bool is_verified = false;
+
+    /* Keys 8 to 15 can store public keys. */
+    if (key_id < 7 || key_id > 15)
+    {
+        return PSA_ERROR_INVALID_ARGUMENT;
+    }
+
+    /* We can only do ECDSA on SHA-256 */
+    if (alg != PSA_ALG_ECDSA(PSA_ALG_SHA_256) && alg != PSA_ALG_ECDSA_ANY)
+    {
+        return PSA_ERROR_NOT_SUPPORTED;
+    }
+
+    if (hash_length != PSA_HASH_SIZE(PSA_ALG_SHA_256))
+    {
+        /* The driver only supports hashes of length 32. */
+        return PSA_ERROR_NOT_SUPPORTED;
+    }
+
+    if (signature_length != ATCA_SIG_SIZE)
+    {
+        /* The driver only supports signatures of length 64. */
+        return PSA_ERROR_NOT_SUPPORTED;
```

Comment:
A signature whose length isn't 64 would be invalid in any implementation, so `INVALID_SIGNATURE` would be the right error code here.